gateway/ln-gateway/src/ng/mod.rs

```diff
@@ -488,7 +467,7 @@ impl GatewayClientModule {
 #[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
 pub enum GatewayClientStateMachines {
     Pay(GatewayPayStateMachine),
-    Receive(GatewayReceiveStateMachine),
+    Receive(InternalTxStateMachine),
```

Comment:
I think `InternalTxStateMachine` is still a bit confusing. Because this will also process external lightning payments into the federation. Can we call it `IncomingStateMachine` or something like that? That covers both the internal tx and external lightning payments.

Reply:
`IncomingStateMachine` makes sense to me from the perspective of the contract being created and funded, but does not seem satisfactory, especially considering all the other SMs are not named per the underlying contracts. Adopting anyways

Reply:
> especially considering all the other SMs are not named per the underlying contracts

This is probably an oversight on my part. In my opinion we should probably rename the `pay` state machine as well.